src/library/psi_library.jl

```diff
@@ -1149,3 +1149,415 @@ function build_two_zone_5_bus(; kwargs...)
     end
     return sys
 end
+
+const COST_PERTURBATION_NOISE = rand(1_000_000)
+
+function _duplicate_system(main_sys::PSY.System, twin_sys::PSY.System, HVDC_line::Bool)
+    names = [
+        "114_SYNC_COND_1",
+        "314_SYNC_COND_1",
+        "313_STORAGE_1",
+        "214_SYNC_COND_1",
+        "212_CSP_1",
+    ]
+    for sys in [main_sys, twin_sys]
+        for d in get_components(
+            x -> get_fuel(x) == ThermalFuels.DISTILLATE_FUEL_OIL,
+            ThermalStandard,
+            sys,
+        )
+            for s in get_services(d)
+                remove_service!(d, s)
+            end
+            remove_component!(sys, d)
+        end
+        for d in PSY.get_components(x -> x.name ∈ names, PSY.Generator, sys)
+            for s in get_services(d)
+                remove_service!(d, s)
+            end
+            remove_component!(sys, d)
+        end
+        for d in
+            get_components(x -> get_fuel(x) == ThermalFuels.NUCLEAR, ThermalStandard, sys)
+            set_must_run!(d, true)
+        end
+    end
+
+    # clear time series
+    PSY.clear_time_series!(twin_sys)
+
+    # change names of the systems
+    PSY.set_name!(main_sys, "main")
+    PSY.set_name!(twin_sys, "twin")
+
+    # change the names of the areas and loadzones first
+    for type_ in [PSY.Area, PSY.LoadZone]
+        for b in PSY.get_components(type_, twin_sys)
+            name_ = PSY.get_name(b)
+            main_comp = PSY.get_component(type_, main_sys, name_)
+            # remove the component
+            PSY.remove_component!(twin_sys, b)
+            # change name
+            PSY.set_name!(b, name_ * "_twin")
+            # define time series container
+            IS.assign_new_uuid!(b)
```

Comment:
This will not work with `psy4`. UUIDs are now used for storage and so you need to pass the system. I think that means we need a sibling branch here.

Reply:
This can be updated when we port PSB to Psy4. We need this system in the short run. @GabrielKS be aware of this.